android/src/main/kotlin/com/example/android_locator_plugin/AndroidLocatorPlugin.kt

```diff
@@ -1,35 +1,180 @@
+            override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+                locationEventSource = events
+            }
+
+            override fun onCancel(arguments: Any?) {
+                locationEventSource = null
+            }
+        }
+
+    private lateinit var accessEventChannel: EventChannel
+    private var accessEventSource: EventChannel.EventSink? = null
+    private var accessStreamHandler: EventChannel.StreamHandler =
+        object : EventChannel.StreamHandler {
+            override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
+                accessEventSource = events
+            }
+
+            override fun onCancel(arguments: Any?) {
+                accessEventSource = null
+            }
+        }
+
+    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        context = flutterPluginBinding.applicationContext
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger, androidLocatorPlugin)
+        channel.setMethodCallHandler(this)
+        accessEventChannel = EventChannel(flutterPluginBinding.binaryMessenger, accessStream)
+        accessEventChannel.setStreamHandler(accessStreamHandler)
+        locationEventChannel = EventChannel(flutterPluginBinding.binaryMessenger, eventLocation)
+        locationEventChannel.setStreamHandler(locationStreamHandler)
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        when (call.method) {
+            initialize -> initializePlugin(result)
+            checkPermission -> checkPermission(result)
+            requestPermission -> requestPermission(result)
+            else -> result.notImplemented()
+        }
+    }
+
+    private fun requestPermission(result: Result) {
+        context.run {
+            activity.apply {
+                requestPermissions(
+                    this,
+                    arrayOf(
+                        Manifest.permission.ACCESS_COARSE_LOCATION,
+                        Manifest.permission.ACCESS_FINE_LOCATION
+                    ),
+                    requestCode
+                )
+            }
+        }
+        result.success(null)
+    }
+
+    private fun checkPermission(result: Result) {
+        val granted = permissionGranted()
+        accessEventSource?.success(if (granted) accessGranted else accessDenied)
+        result.success(granted)
+    }
+
+    private fun initializePlugin(result: Result) {
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(activity)
+        val permission = permissionGranted()
+        accessEventSource?.success(if (permission) pluginInitialized else pluginNeedPermission)
+        if (permission) {
+            locationRequest = LocationRequest.create()
+            locationRequest.interval = locationRequestInterval
+            locationRequest.priority = PRIORITY_HIGH_ACCURACY
```

Comment:
In this case you could use with(locationRequest){ ... }

Reply:
Read about scope functions to accomplish Fran's comment

Reply:
with scope function implemented